src/url-mapper.js

```diff
@@ -4,18 +4,73 @@ export default class UrlMapper {
     this._db = db;
     this._update = update;
     this._map = {};
-    this._mapByNewUrl = {};
-    this._count = 0;
+    this._wildcards = {};
 
-    this.getList((err, mappedUrls) => {
+    this._db.find({}, (err, mappedUrls) => {
       mappedUrls.forEach((mappedUrl) => {
         this._addMemoryCopy(mappedUrl);
       });
+      update(); // Inform UI that mappings have been loaded
     });
   }
 
   get(url) {
-    return this._map[url];
+    const plainUrl = this._map[url];
+    if (plainUrl) {
+      return plainUrl;
+    }
+
+    const wildcardList = [];
+    for (const key in this._wildcards) {
+      if (this._wildcards.hasOwnProperty(key)) {
+        wildcardList.push(this._wildcards[key]);
+      }
+    }
+
+    const urlChunks = url.split('/');
+    const matches = wildcardList.filter(function(mapping) {
+      const chunks = mapping.url.split('/');
+
+      if (urlChunks.length !== chunks.length) {
+        return false;
+      }
+
+      let isMatch = true;
+      chunks.forEach(function(chunk, i) {
+        if (chunk === '*') {
+          return;
+        }
+
+        // TODO handle case where chunks aren't a wildcard, but contain one, e.g. "http://test.com/version*/app.js"
+        // Either properly handle it here, or validate it in UI
+        if (chunk !== urlChunks[i]) {
+          isMatch = false;
+        }
+      });
+      return isMatch;
+    }).sort(function(a, b) {
+      // Use the mapping with the least wildcards
```

Comment:
Should we introduce a function with proper naming, which may help us to get rid of the comment?